src/lib.rs

```diff
@@ -43,18 +46,83 @@ pub struct UPClientZenoh {
     query_map: Arc<Mutex<HashMap<String, Query>>>,
     // Save the callback for RPC response
     rpc_callback_map: Arc<Mutex<HashMap<String, Arc<UtransportListener>>>>,
+    // Used to identify different callback
     callback_counter: AtomicU64,
+    // Source UUri in RPC
+    uuri: UUri,
 }
 
 impl UPClientZenoh {
+    /// Create `UPClientZenoh` by applying the Zenoh configuration.
+    /// The `UUri` will be generated randomly.
+    ///
+    /// # Arguments
+    ///
+    /// * `config` - Zenoh configuration. You can refer to [here](https://github.com/eclipse-zenoh/zenoh/blob/0.10.1-rc/DEFAULT_CONFIG.json5) for more configuration details.
+    ///
     /// # Errors
-    /// Will return `Err` if unable to create Zenoh session
+    /// Will return `Err` if unable to create `UPClientZenoh`
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # async_std::task::block_on(async {
+    ///     use up_client_zenoh::UPClientZenoh;
+    ///     use zenoh::config::Config;
+    ///     let upclient = UPClientZenoh::new(Config::default()).await.unwrap();
+    /// # });
+    /// ```
     pub async fn new(config: Config) -> Result<UPClientZenoh, UStatus> {
+        let uuri = UUri {
+            entity: Some(UEntity {
+                name: "default.entity".to_string(),
+                id: Some(u32::from(rand::random::<u16>())),
+                version_major: Some(1),
+                version_minor: None,
+                ..Default::default()
+            })
+            .into(),
+            ..Default::default()
+        };
+        UPClientZenoh::new_with_uuri(config, uuri).await
+    }
+
+    /// Create `UPClientZenoh` by applying the Zenoh configuration and self-defined `UUri`.
+    ///
+    /// # Arguments
+    ///
+    /// * `config` - Zenoh configuration. You can refer to [here](https://github.com/eclipse-zenoh/zenoh/blob/0.10.1-rc/DEFAULT_CONFIG.json5) for more configuration details.
+    /// * `uuri` - The `UUri` which is put in source while sending RPC request.
+    ///
+    /// # Errors
+    /// Will return `Err` if unable to create `UPClientZenoh`
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # async_std::task::block_on(async {
+    ///     use up_client_zenoh::UPClientZenoh;
+    ///     use up_rust::{UEntity, UUri};
+    ///     use zenoh::config::Config;
+    ///     let uuri = UUri {
+    ///         entity: Some(UEntity {
+    ///             name: "default.entity".to_string(),
+    ///             id: Some(u32::from(rand::random::<u16>())),
+    ///             version_major: Some(1),
+    ///             version_minor: None,
+    ///             ..Default::default()
+    ///         })
+    ///         .into(),
+    ///         ..Default::default()
+    ///     };
+    ///     let upclient = UPClientZenoh::new_with_uuri(Config::default(), uuri).await.unwrap();
+    /// # });
+    /// ```
+    pub async fn new_with_uuri(config: Config, uuri: UUri) -> Result<UPClientZenoh, UStatus> {
```

Comment:
WDYT about trying to initialize the `env_logger` upon construction and just quietly failing if it's already been initialized?

I saw some Zenoh plugin code do this and made sense to me.

Reply:
IMHO, I think `env_logger` is suitable to be used in binary. The library users (no matter the tests, examples, etc.) can decide whether they want to init the logger or how they init the logger by themselves.